services/tables/src/main/java/com/linkedin/openhouse/tables/utils/IntervalToCronConverter.java

```diff
@@ -0,0 +1,75 @@
+package com.linkedin.openhouse.tables.utils;
+
+import com.cronutils.builder.CronBuilder;
+import com.cronutils.model.CronType;
+import com.cronutils.model.definition.CronDefinitionBuilder;
+import com.cronutils.model.field.expression.FieldExpressionFactory;
+import com.linkedin.openhouse.common.exception.RequestValidationFailureException;
+import com.linkedin.openhouse.tables.api.spec.v0.request.components.ReplicationConfig;
+import com.linkedin.openhouse.tables.api.spec.v0.request.components.TimePartitionSpec;
+import java.util.Random;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+/** Utility class for generating a cron schedule given an interval for which a job should run */
+@Slf4j
+@Component
+public class IntervalToCronConverter {
+
+  /**
+   * Public api to generate a cron schedule for a {@link ReplicationConfig} based on a given
+   * interval string in the form 12H, 1D, 2D, 3D.
+   *
+   * @param interval
+   * @return schedule
+   */
+  public static String generateCronExpression(String interval) {
+    if (interval == null || interval.isEmpty()) {
+      String errorMessage = "Replication interval is null or empty";
+      log.error(errorMessage);
+      throw new RequestValidationFailureException(errorMessage);
+    }
+    int count = Integer.parseInt(interval.substring(0, interval.length() - 1));
+    // Generating random hourly and minute intervals for the cron expression
+    int hour = new Random().nextInt(24);
```

Comment:
@chenselena Can you check if 24 hours is a valid hour for quartz cron? I think allowed values range from 0-23.

Reply:
new Random().nextInt(24) will return values [0,24), 24 not included